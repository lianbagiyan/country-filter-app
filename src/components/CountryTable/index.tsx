import React from 'react';
import { CountryTableProps } from "../../types/types";
import './style.scss';

const CountryTable: React.FC<CountryTableProps> = ({ data }) => {
    return (
        <table className="table-container">
            <thead>
            <tr>
                <th>Country Name</th>
                <th>Country Code</th>
            </tr>
            </thead>
            <tbody>
            {data.length ? (
                data.map((country) => (
                    <tr key={country.code}>
                        <td>{country.name}</td>
                        <td>{country.code}</td>
                    </tr>
                ))
            ) : (
                <tr>
                    <td colSpan={2} className="table-container__no-results">
                        No results found.
                    </td>
                </tr>
            )}
            </tbody>
        </table>
    );
};

export default CountryTable;
