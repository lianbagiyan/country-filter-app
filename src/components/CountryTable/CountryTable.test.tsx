import React from 'react';
import { render, screen } from '@testing-library/react';
import CountryTable from './index';

const mockData = [{ name: 'Armenia', code: 'AM' }];

const unsortedMockData = [
  { name: 'Brazil', code: 'BR' },
  { name: 'Armenia', code: 'AM' },
];

describe('CountryTable Component', () => {
  test('renders with data', () => {
    render(<CountryTable data={mockData} />);

    const countryNameElement = screen.getByText('Armenia');
    const countryCodeElement = screen.getByText('AM');

    expect(countryNameElement).toBeInTheDocument();
    expect(countryCodeElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<CountryTable data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  test('renders sorted rows', () => {
    render(<CountryTable data={unsortedMockData} />);

    const firstRow = screen.getByText('Armenia');
    const secondRow = screen.getByText('Brazil');

    expect(firstRow).toBeInTheDocument();
    expect(secondRow).toBeInTheDocument();
  });

  test('renders with no results message', () => {
    render(<CountryTable data={[]} />);

    const noResultsElement = screen.getByText('No results found.');
    expect(noResultsElement).toBeInTheDocument();
  });
});
