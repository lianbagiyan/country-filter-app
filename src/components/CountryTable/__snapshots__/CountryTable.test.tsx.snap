// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryTable Component matches snapshot 1`] = `
<div>
  <table
    class="table-container"
  >
    <thead>
      <tr>
        <th>
          Country Name
        </th>
        <th>
          Country Code
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Armenia
        </td>
        <td>
          AM
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
